packages/app-store/hubspot/lib/CrmService.ts

```diff
@@ -234,49 +178,11 @@ export default class HubspotCalendarService implements Calendar {
     return Promise.reject("Something went wrong when creating a meeting in HubSpot");
   }
 
-  async createEvent(event: CalendarEvent): Promise<NewCalendarEventType> {
+  async createEvent(event: CalendarEvent, contacts: Contact[]): Promise<CrmEvent> {
     const auth = await this.auth;
     await auth.getToken();
-    const contacts = await this.hubspotContactSearch(event);
-    if (contacts.length) {
-      if (contacts.length == event.attendees.length) {
-        // All attendees do exist in HubSpot
-        this.log.debug("contact:search:all", { event, contacts });
-        return await this.handleMeetingCreation(event, contacts);
-      } else {
-        // Some attendees don't exist in HubSpot
-        // Get the existing contacts' email to filter out
-        this.log.debug("contact:search:notAll", { event, contacts });
-        const existingContacts = contacts.map((contact) => contact.properties.email);
-        this.log.debug("contact:filter:existing", { existingContacts });
-        // Get non existing contacts filtering out existing from attendees
-        const nonExistingContacts = event.attendees.filter(
-          (attendee) => !existingContacts.includes(attendee.email)
-        );
-        this.log.debug("contact:filter:nonExisting", { nonExistingContacts });
-        // Only create contacts in HubSpot that were not present in the previous contact search
-        const createContacts = await this.hubspotContactCreate(nonExistingContacts);
-        this.log.debug("contact:created", { createContacts });
-        // Continue with meeting creation and association only when all contacts are present in HubSpot
-        if (createContacts.length) {
-          this.log.debug("contact:creation:ok");
-          return await this.handleMeetingCreation(
-            event,
-            createContacts.concat(contacts) as SimplePublicObjectInput[]
-          );
-        }
-        return Promise.reject("Something went wrong when creating non-existing attendees in HubSpot");
-      }
-    } else {
-      this.log.debug("contact:search:none", { event, contacts });
-      const createContacts = await this.hubspotContactCreate(event.attendees);
-      this.log.debug("contact:created", { createContacts });
-      if (createContacts.length) {
-        this.log.debug("contact:creation:ok");
-        return await this.handleMeetingCreation(event, createContacts as SimplePublicObjectInput[]);
-      }
-    }
-    return Promise.reject("Something went wrong when searching/creating the attendees in HubSpot");
+
+    return await this.handleMeetingCreation(event, contacts);
```

Comment:
I want to keep the methods as atomic as possible. The `createEvent` method should only handle creating events.